polygeist/tools/cgeist/Test/Verification/float16.c

```diff
@@ -19,12 +19,18 @@ _Float16 type(_Float16 arg) {
 }
 
 // CHECK-EXTEND-LABEL:  func.func @arith(%arg0: f16, %arg1: f16, %arg2: f16, %arg3: f16, %arg4: f16) -> f16
-// CHECK-EXTEND-NEXT:     %[[ADD:.*]] = arith.addf %arg0, %arg1 : f16
```

Comment:
Currently, `cgeist` is managing casting on a per-expression basis. I'm doing an effort on unifying this so that every expression promotes/casts operands the same way. I have 2 draft PRs adding this code and enabling this conversion for compound assignment operators.